pkg/graveler/graveler.go

```diff
@@ -1684,31 +1712,43 @@ func (g *KVGraveler) Commit(ctx context.Context, repositoryID RepositoryID, bran
 		if params.SourceMetaRange != nil {
 			commit.MetaRangeID = *params.SourceMetaRange
 		} else {
-			changes, err := g.StagingManager.List(ctx, branch.StagingToken, ListingMaxBatchSize)
+			itrs, err := g.listSealedTokens(ctx, branch)
 			if err != nil {
-				return "", fmt.Errorf("staging list: %w", err)
+				return nil, err
+			}
+			if len(itrs) == 0 { // staging empty
+				return nil, ErrNotFound
+			}
+			if len(itrs) == 1 {
+				commit.MetaRangeID, _, err = g.CommittedManager.Commit(ctx, storageNamespace, branchMetaRangeID, itrs[0])
+			} else {
+				changes := NewCombinedIterator(itrs...)
+				defer changes.Close()
+				commit.MetaRangeID, _, err = g.CommittedManager.Commit(ctx, storageNamespace, branchMetaRangeID, changes)
 			}
```

Comment:
Maybe refactor the combined iterator to accept a single iterator and avoid code branches here?

Reply:
Not sure it should be a part of this pr, I can open an issue